plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadataConfig.java

```diff
@@ -77,8 +77,8 @@ public boolean isComplexJoinPushdownEnabled()
         return complexJoinPushdownEnabled;
     }
 
-    @Deprecated
-    @Config("deprecated.join-pushdown.with-expressions")
+    @Deprecated(forRemoval = true)
```

Comment:
Airlift has a checker that if the setter is `@Deprecated` , the getter is also deprecated.
That's why both are deprecated. Let's mark both with `forRemoval` for consistency.